manifests/instance/dn/add.pp

```diff
@@ -0,0 +1,167 @@
+# Creates the passed DN using the provided paramters
+#
+# NOTE: When calling this defined type as you first set up an instance, you
+# will need to pass in all parameters since the fact will not yet be fully
+# populated.
+#
+# This works by comparing a generated LDIF with the LDIF formatted contents of
+# the target instance.
+#
+# If passing a full LDIF - DO NOT WRAP LINES
+#
+# @option name [String]
+#   A globally unique description of the desired configuration setting
+#
+# @param instance_name
+#   The instance name as passed to `ds389::instance`
+#
+# @param dn
+#   The DN to be created
+#
+# @param objectclass
+#   objectClasses to which the DN belongs
+#
+# @param attrs
+#   Attributes that you wish to set at the time of creation
+#
+# @param content
+#   The full content of the LDIF
+#
+#   * This may only contain *one* entry
+#   * All other parameters will be ignored
+#   * DO NOT add 'changetype: add'
+#
+# @param root_dn
+#   A DN with administrative rights to the directory
+#
+#   * Will be determined automatically if not set
+#
+# @param root_pw_file
+#   A file containing the password for use with ``$root_dn``
+#
+#   * Defaults to `$ds389::config_dir/<usual pw file>`
+#
+# @param port
+#   The port to which to connect
+#
+#   * Has no effect if LDAPI is enabled on the instance
+#   * Will be determined automatically if not set
+#
+# @param force_ldapi
+#   Force the system to use the LDAPI interface
+#
+#   * Generally only useful during bootstrapping
+#
+# @param restart_instance
+#   Whether or not to restart the directory server after applying this item
+#
+define ds389::instance::dn::add (
+  Simplib::Systemd::ServiceName         $instance_name,
```

Comment:
The way this is being used, I do not think the service name is being passed in.  Looks like the $title of a ds389::instance is being used, instead.  Is this cruft from an earlier implementation?

Reply:
The title of the instance is the service name and has to match the syntax allowed by systemd.

It's asserted in `instance.pp` with caveats but this keeps it as close as possible to 'correct'.